### PR TITLE
Feature appframework

### PR DESCRIPTION
**Pull request by sgontla**
Merge feature-appframework to develop as part of the release 1.0.2

### CODE REVIEW

**Review by pdhanoyasplunk — APPROVED** [review]

**Review by jryb — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -105,7 +106,7 @@ executors:
       # Set cluster workers to 5 (5 cluster nodes)
       # NUM_NODES represent number of parallel test executions
       # NUM_WORKERS represent number of nodes in a k8 cluster
-      NUM_NODES: 5 
+      NUM_NODES: 3
```
Comment (gaurav-splunk) [inline .circleci/config.yml:109]: what is this change for?

File: .circleci/config.yml [hunk 2]
```diff
@@ -434,4 +434,4 @@ jobs:
           name: Delete cluster
           command: |
             make cluster-down
-          no_output_timeout: 30m
+          no_output_timeout: 30m
```
Comment (gaurav-splunk) [inline .circleci/config.yml:437]: and this?

File: .circleci/config.yml (hunk 1 above)
Comment (sgontla) [reply .circleci/config.yml:109]: All these are part of the merge, and no manual changes are involved. :-). 
This change came from Param, a couple of days ago as part of a PR.

File: .circleci/config.yml (hunk 2 above)
Comment (sgontla) [reply .circleci/config.yml:437]: Came, as part of the branch merge. No manual changed involved in this PR.

File: .circleci/config.yml (hunk 1 above)
Comment (gaurav-splunk) [reply .circleci/config.yml:109]: @pdhanoyasplunk any specific reason for this change?

Comment (pdhanoyasplunk) [reply .circleci/config.yml:109]: @gaurav-splunk `NUM_NODES` is for parallel execution. Having more parallel test have no benefit and are causing flakiness in test. Hence I reduced the number of parallel test for more reliability in nightly runs.

**Review by gaurav-splunk — APPROVED** [review]

**Comment by pdhanoyasplunk** [comment]
Expected failures in smoke test will pass when splunk:edge has ansible changes.
```
[1] c3appfw test
[1] /home/circleci/project/test/c3/appframework/appframework_test.go:29
[1]   Clustered deployment (C3 - clustered indexer, search head cluster)
[1]   /home/circleci/project/test/c3/appframework/appframework_test.go:233
[1]     smoke, c3, integration, appframework: can deploy a C3 SVA and have apps installed locally on CM and SHC Deployer [It]
[1]     /home/circleci/project/test/c3/appframework/appframework_test.go:234
[1] 
[1]     Timed out after 1500.000s.
[1]     Expected
[1]         <common.Phase>: Pending
[1]     to equal
[1]         <common.Phase>: Ready
[1] 
[1]     /home/circleci/project/test/testenv/verificationutils.go:103
[1] ------------------------------
[1] {"level":"info","ts":1626819302.8885179,"msg":"testenv teardown is skipped!\n","testenv":"5261045-c3appfw-dnj"}
[1] {"level":"info","ts":1626819302.888538,"msg":"testenv teardown is skipped!\n","testenv":"5261045-c3appfw-dnj"}
[1] 
[1] JUnit report was created: /home/circleci/project/test/c3/appframework/c3appfw-dnj_junit.xml
[1] 
[1] 
[1] Summarizing 1 Failure:
[1] 
[1] [Fail] c3appfw test Clustered deployment (C3 - clustered indexer, search head cluster) [It] smoke, c3, integration, appframework: can deploy a C3 SVA and have apps installed locally on CM and SHC Deployer 
[1] /home/circleci/project/test/testenv/verificationutils.go:103
[1] 
[1] Ran 1 of 2 Specs in 2022.382 seconds
[1] FAIL! -- 0 Passed | 1 Failed | 0 Pending | 1 Skipped
```

**Pull request merged**